src/component/component/index.js

```diff
@@ -137,19 +137,31 @@ export class Component extends BaseComponent {
         // Register all of the drivers for instantiating components. The model used is -- there's a standard javascript
         // way of rendering a component, then each other technology (e.g. react) needs to hook into that interface.
         // This makes us a little more pluggable and loosely coupled.
-
+        this.driverResults = {};
```

Comment:
In your second example:

```
imports: [MyLoginXComponent
   .driver('angular2', { Component, NgModule, ElementRef, NgZone, BrowserModule})
  .driverResults
  .angular2
  .module]
```

Providing we return the autogenerated angular2 module in `register()`, this should be:

```
imports: [
    MyLoginXComponent.driver('angular2', {
        Component, NgModule, ElementRef, NgZone, BrowserModule
    })
]
```

The fact that the angular 1 driver returns the xcomponent in `register()` is a mistake -- it doesn't add any value. It should probably just return undefined. But for angular 2 the most useful thing for us to return is the module we create.